pkg/tsdb/legacydata/service/service_test.go

```diff
@@ -57,6 +57,39 @@ func TestHandleRequest(t *testing.T) {
 	})
 }
 
+func Test_generateRequest(t *testing.T) {
+	t.Run("Should attach custom headers to request if present", func(t *testing.T) {
+		jsonData := simplejson.New()
+		jsonData.Set(headerName+"testOne", "x-test-one")
```

Comment:
Would it be better to check that the suffix must be an integer like in ngalert?

Reply:
See the above comment - I don't think we have a guarantee in our API that it will remain an integer.